packages/api/src/tasks/providers/default.ts

```diff
@@ -21,16 +112,57 @@ export default class DefaultAlertProvider implements AlertProvider {
   }
 
   async getAlertTasks(): Promise<AlertTask[]> {
+    const groupedTasks = new Map<string, AlertTask>();
     const alerts = await Alert.find({});
-    if (config.IS_LOCAL_APP_MODE) {
-      alerts.forEach(_alert => {
+    for (const alert of alerts) {
+      if (!alert.source) {
+        logger.error(`alert does not have a source: alertId=${alert._id}`);
+        continue;
+      }
+
+      if (config.IS_LOCAL_APP_MODE) {
         // The id is the 12 character string `_local_team_', which will become an ObjectId
         // as the ASCII hex values, so 5f6c6f63616c5f7465616d5f.
-        _alert.team = new mongoose.Types.ObjectId(LOCAL_APP_TEAM.id);
-      });
+        alert.team = new mongoose.Types.ObjectId(LOCAL_APP_TEAM.id);
+      }
+
+      let conn: IConnection | undefined;
+      let details: AlertDetails | undefined;
+      switch (alert.source) {
+        case AlertSource.SAVED_SEARCH:
+          [conn, details] = await getSavedSearchDetails(alert);
+          break;
+
+        case AlertSource.TILE:
+          [conn, details] = await getTileDetails(alert);
+          break;
+
+        default:
+          logger.error(
+            `unsupported source: alertId=${alert._id}, source=${alert.source}`,
+          );
+          continue;
+      }
+
+      if (!details) {
+        logger.error(`failed to fetch alert details: alertId=${alert._id}`);
+        continue;
+      }
+
+      if (!conn) {
+        logger.error(`failed to fetch alert connection: alertId=${alert._id}`);
+        continue;
+      }
+
+      const k = conn._id.toString();
+      if (!groupedTasks.has(k)) {
+        groupedTasks.set(k, { alerts: [], conn });
+      }
+      groupedTasks.get(k)?.alerts.push(details);
```

Comment:
nit: we should probably throw if get(k) is nullish

Reply:
Sure, we could. The if block above this line is checking the map and populate the entry if the key is missing. The `?` is needed because we can't express that check in the type system. `Map` in JS is just limited in its API.

To comapre, Rust would handle this with a single line that allows expression of the intent without undefined values anywhere:
```
groupedTasks.entry(k)
   .and_modify(|e| e.alerts.push(details))
   .or_insert(AlertDetails::new());
```

Reply:
I can see a world where if we want to defensively program this, that we'd actually wrap every iteration here with a try/catch so that one alert will not crash the entire process and result in all alerts being skipped. The same comment can probably be made to other parts of the loop as well.

For this specific issue alone - I agree, I don't think it's fully necessary, but can see how if we want to program defensively we'd add an extra check in case someone breaks this in the future.

That being said - I think a general try/catch is the better idea here.

Reply:
Reworked this loop a bit, using the idea that each alert should be in a `try...catch`. There is now a separate function that does all the work for a single alert object. If the `get(k)` from map fails, it throws an `Error`. The call to that function is wrapped in a `try...catch` that doesn't stop the iteration of the loop, just logs.